cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala

```diff
@@ -232,29 +259,40 @@ object AnalyseExecutor {
     }
   }
 
+  def isValidTool(tool: String): Boolean = {
```

Comment:
Why do we need to search tools twice and why do we need to match metrics and duplication manually.
I think looking normally for tools should be enough since it should throw an error if cannot find the tool for name.